### ISSUE TITLE
Adding Home Assistant Calendar results in -null in config

### ISSUE THREAD

**Issue by mostlychris** [totaldebug/atomic-calendar-revive#1021]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After upgrading Home Assistant to version 2023.5, an error stating the Google Calendar could not be loaded and no other calendars showed either.  This is a red herring as the issue is that the UI is adding a '-null' entity value for the built in Home Assistant calendar.

### Expected behaviour

All of the calendar's (Google, Home Assistant, etc) events as configured.

### Steps To Reproduce

Go the the UI, modify any settings. This results in a '-null' option being added for the Home Assistant calendar entity instead of the actual entity. Editing the yaml directly and adding the entity resolves the issue.

### Card Version

7.3.0

### Home Assistant Version

2023.5.x

### Configuration

```shell
- type: custom:atomic-calendar-revive
    entities:
      - entity: calendar.texas_a_m_aggies
        name: Texas a&m aggies
      - entity: calendar.{gmail calendar}_gmail_com
        name: Allyson Calendar
        color: yellow
      - entity: calendar.main_calendar
        color: orange
    firstDayOfWeek: 1
    maxDaysToShow: 7
    refreshInterval: 300
    showDeclined: false
    showDate: false
    hideFinishedEvents: true
    showRelativeTime: true
    calShowDescription: false
    showLastCalendarWeek: false
    dimFinishedEvents: true
    showFullDayProgress: true
    showEventIcon: true
    showNoEventsForToday: false
    showWeekDay: true
    showCurrentEventLine: false
    showMonth: false
    showHiddenText: false
    sortByStartTime: false
    hideDuplicates: false
    showDescription: false
    showProgressBar: false
    disableEventLink: true
    disableLocationLink: true
    showLoader: false
    showLocation: false
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bkr1969** [comment]
Here is my code, though it shouldn't be necessary for reporting this problem. Any time you try to edit the caledar, a -null is inserted and invalidates the visual editor.

type: custom:atomic-calendar-revive
name: Calendar Card
entities:
  - null
  - entity: calendar.anylist_meal_plan
    name: Anylist meal plan
    color: grey
    icon: mdi:silverware-fork-knife
  - entity: calendar.ohio_state_buckeyes
    name: Ohio state buckeyes
    icon: mdi:football
    color: maroon
enableModeChange: true
defaultMode: Calendar
calShowDescription: true
showLastCalendarWeek: false
disableCalEventLink: false
disableCalLink: true
firstDayOfWeek: 0
maxDaysToShow: 10
refreshInterval: 60
showDate: true
showRelativeTime: true
showMultiDay: false
hideDuplicates: true
sortByStartTime: true
showCurrentEventLine: true
calEventTime: true
showMonth: true
showFullDayProgress: true
showWeekDay: false
showEventIcon: true
showCalendarName: true
showHiddenText: false
showDatePerEvent: false
disableCalLocationLink: false
showMultiDayEventParts: true
hideFinishedEvents: false
untilText: 'true'

**Comment by bkr1969** [comment]
Just realized something else.  The ```-null``` is not just added, but it actually replaces one of the calendars added.  Bizarre.

**Comment by marksie1988** [comment]
Hi guys, could you tell me what you are doing when this happens? 

I have tried adding a calendar and it works without any issues i dont get a null like you are seeing

**Comment by bkr1969** [comment]
Just adding one or more calendars to an existing config.

**Comment by marksie1988** [comment]
> Just adding one or more calendars to an existing config. Thanks for responding!

Ok tha KS, I can't get it to do this, I'll do a little more playing see what I can find. Do you know is it a caldav or Google calendar that it's happening to?

**Comment by bkr1969** [comment]
caldav

**Comment by mostlychris** [comment]
For me if I do anything in the UI, it causes the -null.

**Comment by bkr1969** [comment]
Suddenly all of my caldav calendars are missing.

**Comment by bkr1969** [comment]
Doesn't help that as of today (5-15-23) with update to Core 2023.5.3, caldav seems to have stopped working.

**Comment by bkr1969** [comment]
OK...just trying to provide as much info as possible.  Not sure about before, but after core 2023.5.3, the ```-null``` appears in place of any caldav calendar you try to add. if there are none and the visual editor is useable, as soon as you turn on a caldav calendar you are booted to the yaml code and the ```-null``` is already there in place of the calendar just turned on.  If you add it in the yaml, it changes to ```-null``` as soon as you save.

**Comment by bkr1969** [comment]
```
Logger: frontend.js.latest.202305033
Source: components/system_log/__init__.py:257
First occurred: May 16, 2023 at 07:43:31 (66 occurrences)
Last logged: May 16, 2023 at 07:44:06

http://xxx.xx.x.xxx:8123/hacsfiles/atomic-calendar-revive/atomic-calendar-revive.js?hacstag=246549747720:54:145885 Uncaught TypeError: Cannot read properties of undefined (reading 'entity')
```

**Comment by jamespo** [comment]
This also happens for Google calendars for me, can add in yaml, not gui. In latest current version of HA & plug-in.

**Comment by marksie1988** [comment]
this should be resolved as part of #1059
